src/shared_modules/indexer_connector/src/monitoring.hpp

```diff
@@ -86,96 +172,60 @@ class Monitoring final
                     {
                         // Check the health of the servers.
-                        for (auto& value : m_values)
+                        for (const auto& [serverAddress, _] : m_servers)
                         {
-                            // Get the health of the server.
-                            HTTPRequest::instance().get(
-                                HttpURL(value.first + "/_cat/health?v"),
-                                [&](std::string response)
-                                {
-                                    // Remove the tabs and double spaces.
-                                    Utils::replaceAll(response, "\t", " ");
-                                    response = Utils::trimRepeated(response, ' ');
-                                    // Split the response by rows.
-                                    const auto rows {Utils::split(response, '\n')};
-
-                                    // Check if the response has the expected number of rows.
-                                    if (HealthCheckRows::SIZE == rows.size())
-                                    {
-                                        // Split the data row by spaces.
-                                        const auto fields {Utils::split(rows.at(HealthCheckRows::DATA), ' ')};
-
-                                        // Check if the response has the expected number of columns and if the status is
-                                        // green.
-                                        if (fields.size() == HealthCheckColumns::SIZE &&
-                                            fields.at(HealthCheckColumns::STATUS).compare("green") == 0)
-                                        {
-                                            value.second = true;
-                                        }
-                                        else
-                                        {
-                                            value.second = false;
-                                        }
-                                    }
-                                    else
-                                    {
-                                        value.second = false; // LCOV_EXCL_LINE
-                                    }
-                                },
-                                [&](const std::string& error, const long /*statusCode*/) { value.second = false; },
-                                "",
-                                headers,
-                                secureCommunication);
+                            healthCheck(serverAddress, authentication);
                         }
                     }
                 }
             });
+
+        // Wait for the initialization to complete or the timeout to expire.
+        std::unique_lock initLock(m_initMutex);
+        m_initCondition.wait_for(
+            initLock, std::chrono::seconds(INITIALIZATION_TIMEOUT), [this]() { return m_initialized; });
```

Comment:
It is better to avoid this logic, which is complicated and can also cause certain exceptions if the timeout occurs.


Let's update http request to have the timeout benefits that it has.


https://github.com/wazuh/wazuh-http-request/blob/5807c8beaf5cf15fddb519253acef25b661353c3/src/HTTPRequest.cpp#L46


It is important that we initialize the vector to false for all servers, to avoid an access exception in the isAvailable() method

Reply:
Making this change will require a lot of code tweaking, as the interface for http-request changed on this PR
- https://github.com/wazuh/wazuh-http-request/commit/ce95de12171c55aec27393f869ff9035fdd8d63f

Do you want me to proceed?

Reply:
yes, I want that.

Reply:
On it then!

Reply:
The problem with not waiting for the server to initialize correctly is that servers will start in a false state for a time even if they are in green (givin error/warn logs). I will see how I can circumvent this